README.md

```diff
@@ -51,17 +51,17 @@ In witness automata, the GraphML edges represent state transitions.
-| enterFunction | *Valid values:* Function name <br /> The name of the function that is entered via this transition. Assuming a function stack, this pushes the function onto the stack. If you use this data node type, you also must use the type ``returnFromFunction``. When ``assumption.scope`` is not given, the witness validator may use this information to qualify  variable names used in ``assumption`` data tags. The path is considered to stay in the specified function until another transition is annotated with this data node for another function or a transition annotated with ``returnFromFunction``, telling the validator that the path continues in the previous function on the stack.  | Yes | Yes |
-| returnFromFunction | *Valid values:* Function name <br /> The name of the function is exited via this transition. Assuming a function stack, this name must match the name of the function popped from the function stack. If you use this data node type, you also must use the type ``enterFunction``. See ``enterFunction`` for more information. | Yes | Yes |
+| assumption | *Valid values:* Expression that evaluates to a value of the equivalent of a boolean type in the programming language of the verification task that **may** consist of conjunctions or disjunctions, but not function calls <br /> ``assumption`` is used to specify a state-space guard for a transition. The value of a ``data`` element with this key must be an expression that evaluates to a value of the equivalent of a boolean type in the programming language of the verification task, e.g. for C programs a C expression that evaluates to a value of the C type ``int`` (used as boolean). The expression may consist of conjunctions or disjunctions, but not function calls. Local variables that have the same name as global variables or local variables of other functions can be qualified by using a ``data`` element with the ``assumption.scope`` key. All variables used in the expression must appear in the program source code, with the exception of the variable ``\result``, which represents the return value of a function identified by the ``data`` element with the key ``assumption.resultfunction`` after a function-return operation on a CFA edge matched by this transition. If the ``\result`` variable is used, the name of the corresponding function must be provided using a ``data`` element with the ``assumption.resultfunction`` key. If a transition does not have a ``data`` element with this key, a consumer shall assume that the state-space guard of this transition is ``true``. In correctness witnesses, for each state, the disjunction of all state-space guards leaving that state must be ``true``, i.e., while state-space guards can be used to split the state space in correctness witnesses, they may not be used to restrict it. The ``attr.type`` attribute of this key is ``string``. | Yes | No |
```

Comment:
May it also consist of a mixture of conjunctions and disjunctions?

There is a duplicate in the text, it is mentioned twice that it may consist of con/disjunction but not function calls:

> Expression that evaluates to a value of the equivalent of a boolean type in the programming language of the verification task that **may** consist of conjunctions or disjunctions, but not function calls

vs.

> The expression may consist of conjunctions or disjunctions, but not function calls

The first occurrence is a bit confusing, how can the value of the expression consist of conjunctions and disjunctions, didn't you mean that the expression itself can consist of those? I would remove the first mention of the constraints on the expression ("that **may** consist of conjunctions or disjunctions, but not function calls") and keep the second

Reply:
> May it also consist of a mixture of conjunctions and disjunctions?
"or" includes "and".
> There is a duplicate in the text, it is mentioned twice that it may consist of con/disjunction but not function calls
Once in the "valid values" column and once in the description. It makes sense in both columns, no? As long as there is no contradiction between both occurrences, I would prefer to keep them both.
> The first occurrence is a bit confusing, how can the value of the expression consist of conjunctions and disjunctions, didn't you mean that the expression itself can consist of those? I would remove the first mention of the constraints on the expression ("that may consist of conjunctions or disjunctions, but not function calls") and keep the second

Agreed, the wording is not only confusing, it is incorrect and I will fix it.